src/library/scala/collection/mutable/Seq.scala

```diff
@@ -42,6 +42,45 @@ trait SeqOps[A, +CC[_], +C <: AnyRef]
     */
   @throws[IndexOutOfBoundsException]
   def update(idx: Int, elem: A): Unit
+
+  /** Sorts this $coll in place according to an Ordering.
+    *
+    * @see [[scala.collection.SeqOps.sorted]]
+    *
+    * @param  ord the ordering to be used to compare elements.
+    * @return modified input $coll sorted according to the ordering `ord`.
+    */
+  def sortInPlace[B >: A](implicit ord: Ordering[B]): this.type = {
+    val len = this.length
+    if (len > 1) {
+      val arr = new Array[AnyRef](len)
+      var i = 0
+      for (x <- this) {
+        arr(i) = x.asInstanceOf[AnyRef]
+        i += 1
+      }
+      java.util.Arrays.sort(arr, ord.asInstanceOf[Ordering[Object]])
+      i = 0
+      while (i < arr.length) {
+        update(i, arr(i).asInstanceOf[A])
```

Comment:
This requires an efficient `update` method.  Are we certain that all instances of `mutable.Seq` have one that runs in at worst `O(log n)`?

Reply:
@Ichoran if I am not mistaken, this is the status of the `update` method speed for all instances of `mutable.Seq`:
* :white_check_mark: `mutable.ArrayBuffer`
* :white_check_mark: `mutable.ArrayDeque`
* :white_check_mark: `mutable.ArraySeq`
* :white_check_mark: `mutable.StringBuilder`
* :x: `mutable.ListBuffer`
* :x: `mutable.UnrolledBufer`